### PR TITLE
feat: add type for response

### PR DESCRIPTION
**Pull request by Zxilly**
and other chore fixes.

I've added some configurations to help the unit tests run properly, but I haven't really executed all of them in their entirety. This is because there is a 2s wait per request and this library doesn't have a proper flow limiting strategy right now.

### CODE REVIEW

**Comment by Zxilly** [comment]
@kerolloz please review this

**Comment by kerolloz** [comment]
Hello @Zxilly 
Sorry for the late review.

File: package.json
Comment (kerolloz) [inline package.json:1]: could you please bump the version to 2.0.2 so that we can release a new version to NPM

Comment (Zxilly) [reply package.json:1]: updated

**Review by kerolloz — APPROVED** [review]

**Pull request merged**